tests/foreman/api/test_usergroup.py

```diff
@@ -0,0 +1,275 @@
+"""Unit tests for the ``usergroups`` paths.
+
+Each ``APITestCase`` subclass tests a single URL. A full list of URLs to be
+tested can be found here:
+http://theforeman.org/api/1.11/apidoc/v2/usergroups.html
+
+"""
+from fauxfactory import gen_string
+from nailgun import entities
+from random import randint
+from requests.exceptions import HTTPError
+from robottelo.datafactory import (
+    invalid_values_list,
+    valid_data_list,
+    valid_usernames_list,
+)
+from robottelo.decorators import tier1, tier2
+from robottelo.test import APITestCase
+
+
+class UserGroupTestCase(APITestCase):
+    """Tests for the ``usergroups`` path."""
+
+    @tier1
+    def test_positive_create_with_name(self):
+        """Create new user group using different valid names
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                user_group = entities.UserGroup(name=name).create()
+                self.assertEqual(user_group.name, name)
+
+    @tier1
+    def test_positive_create_with_user(self):
+        """Create new user group using valid user attached to that group.
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        for login in valid_usernames_list():
+            with self.subTest(login):
+                user = entities.User(login=login).create()
+                user_group = entities.UserGroup(user=[user]).create()
+                self.assertEqual(len(user_group.user), 1)
+                self.assertEqual(user_group.user[0].read().login, login)
+
+    @tier1
+    def test_positive_create_with_users(self):
+        """Create new user group using multiple users attached to that group.
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully and contains all expected
+        users.
+        """
+        users = [entities.User().create() for _ in range(randint(3, 5))]
+        user_group = entities.UserGroup(user=users).create()
+        self.assertEqual(
+            set(user.login for user in users),
```

Comment:
I would not use set here, but a sorted list.

Set can mask a failure when an user is added two times for example.